lib/service/ufind_service.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:we_find/extensions/string_extensions.dart';
+import 'package:we_find/extensions/xml_extensions.dart';
+import 'package:we_find/model/model.dart';
+
+const authority = "m1-ufind.univie.ac.at";
+
+/// Returns the root [StudyModule] based on the specified semester [when]
```

Comment:
Nitpick, but let's rename `when` to something else. `when` sounds like a reserved keyword. Let's use `semester` or `moduleDate` or something similar.